sim/src/make/activity_model.rs

```diff
@@ -2,114 +2,264 @@ use crate::{
+                // they came.
+                // TODO: instead, if it's not a two-way border, we should find an intersection
+                // an incoming border "near" the outgoing border, to allow a broader set of
+                // realistic options.
+                if border.is_incoming_border() {
+                    return TripEndpoint::Border(border.id, None);
+                }
+            }
+            debug_assert!(
+                false,
+                "failed to find a 2 way border in a reasonable time. Degenerate map?"
+            );
+            TripEndpoint::Border(all_outgoing_borders.choose(&mut tmp_rng).unwrap().id, None)
+        };
+
+        let person_params = (0..num_trips)
+            .map(|_| {
+                let (is_local_resident, is_local_worker) = (
+                    rng.gen_bool(prob_local_resident),
+                    rng.gen_bool(prob_local_worker),
+                );
+                let home = if is_local_resident {
+                    if let Some(residence) = residents.pop() {
+                        TripEndpoint::Bldg(residence)
+                    } else {
+                        warn!(
+                            "unexpectedly out of residential capacity, falling back to off-map \
+                             residence"
+                        );
+                        get_commuter_border()
+                    }
+                } else {
+                    get_commuter_border()
+                };
+
+                let work = if is_local_worker {
+                    if let Some(workplace) = workers.pop() {
+                        TripEndpoint::Bldg(workplace)
+                    } else {
+                        warn!(
+                            "unexpectedly out of workplace capacity, falling back to off-map \
+                             workplace"
+                        );
+                        get_commuter_border()
+                    }
+                } else {
+                    get_commuter_border()
+                };
+
+                match (&home, &work) {
+                    (TripEndpoint::Bldg(_), TripEndpoint::Bldg(_)) => {
+                        num_trips_local += 1;
+                    }
+                    (TripEndpoint::Bldg(_), TripEndpoint::Border(_, _)) => {
+                        num_trips_commuting_out += 1;
+                    }
+                    (TripEndpoint::Border(_, _), TripEndpoint::Bldg(_)) => {
+                        num_trips_commuting_in += 1;
+                    }
+                    (TripEndpoint::Border(_, _), TripEndpoint::Border(_, _)) => {
+                        num_trips_passthru += 1;
+                    }
+                };
+
+                (home, work, abstutil::fork_rng(rng))
+            })
+            .collect();
+
+        timer
             .parallelize(
-                "create border trips",
+                "create people: building PersonSpec from endpoints",
```

Comment:
nit: s/building/creating or making/, because the first word is a bit overloaded in this context :P